release-notes-1.2.0.md

```diff
@@ -0,0 +1,751 @@
+
+This release introduces expression-bodied functions whose body is a single expression. The expression function body takes the form `=> E;`, where `E` is any expression. It is equivalent to the block function body, `{ return E; }`. The following is an example where a `Person` record is mapped to an `Employee` record using an expression-bodied function.
+
+```ballerina
+function toEmployee(Person p, string pos) returns Employee => {
+    name: p.fname + " " + p.lname,
+    designation: pos
+};
+```
+
+### Let expressions
+
+This release introduces let expression. It takes the form `let T B = E1 in E2`, where `E1` is evaluated resulting in a value `v`. The typed binding pattern `T B` is matched to `v`, causing assignments to the variables occurring in `B`. Then `E2` is evaluated with those variables in scope; the resulting value is the result of the let expression.
+
+```ballerina
+const int globalVar = 2;
+public function main() {
+    int b = let int x = 4, int y = 3 in x * y * globalVar; // b = 4 * 3 * 2 = 24
+}
+```
+
+### Improved mapping constructor syntax
+
+#### Spread operator
+
+A mapping constructor expression can now have a spread field. A spread field can be used with another mapping value `V` to include all of the fields in `V` when creating the new mapping value.
+
+```ballerina
+type Foo record {|
+    string s;
+    decimal d;
+|};
+
+type Bar record {|
+    string s;
+    int i;
+    decimal...;
+|};
+
+public function main() {
+    Foo f = {s: "test str", d: 1.0};
+
+    // Spread field `...f` spreads the fields of `f` 
+    // when creating the `Bar` value `b`. The required 
+    // field `s` is provided via `f`. 
+    // `b` will contain 3 fields; `s`, `i`, and `d`.
+    Bar b = {i: 1, ...f};
+}
+```
+
+#### Variable names as fields of mapping constructors
+
+A mapping constructor expression can also contain just a variable name (`foo`) as a field. This is equivalent to the key-value pair field `foo: foo`. The name of the variable is considered the key while the variable reference is considered the expression.
+
+```ballerina
+type Employee record {|
+    string name;
+    string department;
+|};
+
+public function main() {
+    string department = "Finance";
+
+    // The variable-name `department` is 
+    // used as a field here.
+    Employee e = {name: "Jo", department};
+}
+``` 
+### Improvements to metadata
+
+#### Deprecation syntax
+
+You can now mark type definitions, functions, object methods, and constants as deprecated using the `@deprecated` annotation. The compiler will generate warnings if a you use a deprecated construct. If the deprecated construct contains documentation, you need to add some additional bit of documentation called `Deprecated` documentation. The `Deprecated` documentation should ideally include details on why the construct was deprecated and suitable alternate options, which should be used instead.
+
+```ballerina
+# Creates and returns a `Baz` object.
+#
+# # Deprecated
```

Comment:
@hasithaa should we remove the Deprecated documentation from this example?

Reply:
and the content?

Reply:
let's keep this. We are going to fix the docgen in the next patch release.